src/SprykerEco/Zed/AmazonQuicksight/Dependency/External/AmazonQuicksightToAwsQuicksightClientAdapter.php

```diff
@@ -0,0 +1,171 @@
+    {
+        return $this->quicksightClient->deleteUserByPrincipalId($deleteUserRequestData);
+    }
+
+    /**
+     * @param array<string, mixed> $deleteUserRequestData
+     *
+     * @return \Aws\ResultInterface<string, mixed>
+     */
+    public function deleteUser(array $deleteUserRequestData): ResultInterface
+    {
+        return $this->quicksightClient->deleteUser($deleteUserRequestData);
+    }
+
+    /**
+     * @param array<string, mixed> $listUsersRequestData
+     *
+     * @return \Aws\ResultInterface<string, mixed>
+     */
+    public function listUsers(array $listUsersRequestData): ResultInterface
+    {
+        return $this->quicksightClient->listUsers($listUsersRequestData);
+    }
+
+    /**
+     * @param array<string, mixed> $updateUserRequestData
+     *
+     * @return \Aws\ResultInterface<string, mixed>
+     */
+    public function updateUser(array $updateUserRequestData): ResultInterface
+    {
+        return $this->quicksightClient->updateUser($updateUserRequestData);
+    }
+
+    /**
+     * @param array<string, mixed> $startAssetBundleImportJobRequestData
+     *
+     * @return \Aws\ResultInterface<string, mixed>
+     */
+    public function startAssetBundleImportJob(array $startAssetBundleImportJobRequestData): ResultInterface
+    {
+        return $this->quicksightClient->startAssetBundleImportJob($startAssetBundleImportJobRequestData);
+    }
+
+    /**
+     * @param array<string, mixed> $describeAssetBundleImportJobRequestData
+     *
+     * @return \Aws\ResultInterface<string, mixed>
+     */
+    public function describeAssetBundleImportJob(array $describeAssetBundleImportJobRequestData): ResultInterface
+    {
+        return $this->quicksightClient->describeAssetBundleImportJob($describeAssetBundleImportJobRequestData);
+    }
+
+    /**
+     * @link https://docs.aws.amazon.com/aws-sdk-php/v3/api/class-Aws.AwsClient.html#method___construct
+     *
+     * @param \SprykerEco\Zed\AmazonQuicksight\AmazonQuicksightConfig $config
+     *
+     * @return array<string, mixed>
+     */
+    protected function getQuicksightClientConfiguration(AmazonQuicksightConfig $config): array
+    {
+        $quicksightClientConfiguration = [
+            'region' => $config->getAwsRegion(),
+            'version' => $config->getQuicksightApiVersion(),
+        ];
+
+        $awsCredentialsKey = $config->findAwsCredentialsKey();
+        $awsCredentialsSecret = $config->findAwsCredentialsSecret();
+        $awsCredentialsToken = $config->findAwsCredentialsToken();
+
+        if ($awsCredentialsKey && $awsCredentialsSecret && $awsCredentialsToken) {
+            $quicksightClientConfiguration['credentials'] = new Credentials(
+                $awsCredentialsKey,
+                $awsCredentialsSecret,
+                $awsCredentialsToken,
+            );
+
+            return $quicksightClientConfiguration;
+        }
+
+        $quicksightAssumedRoleArn = $config->getQuicksightAssumedRoleArn();
```

Comment:
you don't need this variable to send it to the next method

Reply:
we need it to check the existence, refactored to pass to another method